### DISCUSSION TITLE
allow uninterrupted display

### DISCUSSION THREAD

**Discussion by ousia** [ruffle-rs/ruffle#10211] [Q&A] [answered]
I use Flash for presentations, which require no constant user input, only watching them.

This differs from other kind of animations, such as games (where user input is required).

The minimal presentation could be https://www.improve-presentations.tk (more elaborate examples would be https://www.free-culture.tk/ or https://www.free-culture.tk/original/).

I tested one of these presentations once on an Android phone with Chrome (I guess an iPhone wouldn’t be different) and the main problem was that the the phone (or the browser) sent the screen to energy-saving mode when it wasn’t receiving user input.

Sorry, I know that my wording is poor,  but is there any way to allow uninterrupted display on such devices?

Let me know whether my explanation is not clear. Many thanks for your help.

**Comment by n0samu** [comment, marked as answer]
Don't worry, your question is perfectly clear. We should be able to support this functionality in Android Chrome with the [Wake Lock API](https://developer.chrome.com/articles/wake-lock/), if someone wants to work on implementing it. It seems like a low-priority task though.

**Reply by ousia** [reply]
Many thanks for your reply, @n0samu.

MDN seems to [document the Wake Lock API](https://developer.mozilla.org/en-US/docs/Web/API/Screen_Wake_Lock_API), but Firefox doesn’t include it.
